can/interfaces/socketcan/socketcan.py

```diff
@@ -286,7 +287,18 @@ def send_bcm(bcm_socket: socket.socket, data: bytes) -> int:
             raise e
 
 
-def _add_flags_to_can_id(message: Message) -> int:
+GLOBAL_TASK_ID = 0
+TASK_LOCK = threading.Lock()
```

Comment:
Instead of making this global, perhaps consider storing this on the `Bus` context since we map 1:1 to a SocketCAN bus. This is because the kernel code matches based on the combination of:

1. Task ID (`op->can_id`)
2. Interface (`op->ifindex`)
3. Flags (`op->flags`)

```c
/*
 * helpers for bcm_op handling: find & delete bcm [rx|tx] op elements
 */
static struct bcm_op *bcm_find_op(struct list_head *ops,
				  struct bcm_msg_head *mh, int ifindex)
{
	struct bcm_op *op;

	list_for_each_entry(op, ops, list) {
		if ((op->can_id == mh->can_id) && (op->ifindex == ifindex) &&
		    (op->flags & CAN_FD_FRAME) == (mh->flags & CAN_FD_FRAME))
			return op;
	}

	return NULL;
}
```

(Taken from the [5.5.7 kernel](https://elixir.bootlin.com/linux/v5.5.7/source/net/can/bcm.c#L700))

I think this should be doable if we expand the `socketcan.CyclicTask` to take an ID argument in `__init__`. Thoughts?

Reply:
This looks good to me. I am thinking about adding a member function to `SocketcanBus` class that returns next task ID, incrementing old one. The result of that function passed to `CyclicSendTask::__init__` as an additional argument. I will try that.